custom_components/tesla_custom/button.py

```diff
@@ -1,84 +1,147 @@
+        car: TeslaCar,
+        coordinator: TeslaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize flash light entity."""
+        super().__init__(hass, car, coordinator)
+        self.type = "flash lights"
+        self._attr_icon = "mdi:car-light-high"
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self._car.flash_lights()
+
+
+class TeslaCarWakeUp(TeslaCarEntity, ButtonEntity):
+    """Representation of a Tesla car wake up button"""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        car: TeslaCar,
+        coordinator: TeslaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize wake up button."""
+        super().__init__(hass, car, coordinator)
+        self.type = "wake up"
+        self._attr_icon = "mdi:moon-waning-crescent"
+        self._attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self._car.wake_up()
+
+
+class TeslaCarForceDataUpdate(TeslaCarEntity, ButtonEntity):
+    """Representation of a Tesla car force data update button."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        car: TeslaCar,
+        coordinator: TeslaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize force data update button."""
+        super().__init__(hass, car, coordinator)
+        self.type = "force data update"
+        self._attr_icon = "mdi:database-sync"
+        self._attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self.update_controller(wake_if_asleep=True, force=True)
 
-    def __init__(self, tesla_device, coordinator):
-        """Initialise the button."""
-        super().__init__(tesla_device, coordinator)
-        self.controller = coordinator.controller
+    @property
+    def available(self) -> bool:
+        """Return True."""
+        return True
+
+
+class TeslaCarTriggerHomelink(TeslaCarEntity, ButtonEntity):
+    """Representation of a Tesla car Homelink button."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        car: TeslaCar,
+        coordinator: TeslaDataUpdateCoordinator,
+    ) -> None:
+        """Initialise Homelink button."""
+        super().__init__(hass, car, coordinator)
+        self.type = "trigger homelink"
+        self._attr_icon = "mdi:garage"
         self.__waiting = False
 
     @property
     def available(self) -> bool:
         """Return True if entity is available."""
-        return (
-            super().available and self.tesla_device.available() and not self.__waiting
-        )
+        return super().available and not self.__waiting
 
-    @TeslaDevice.Decorators.check_for_reauth
     async def async_press(self, **kwargs):
         """Send the command."""
-        _LOGGER.debug("Trigger homelink: %s", self.name)
         self.__waiting = True
```

Comment:
@carleeno I'm helping with the rewrite for this integration and had a question about this addition you made for HomeLink. I noticed you added `self.__waiting` which looks like it's used only for setting the `available` property, so when the button is pressed, the button becomes unavailable in HA. Just curious as to why that was needed - does it cause issues if you press the HomeLink button again when, for example, a garage door is in transit? I thought it would just send the command again.